public/pages/ConfigureForecastModel/components/AdvancedSettings/AdvancedSettings.tsx

```diff
@@ -0,0 +1,337 @@
+                    <EuiBadge color="default" className="unit-badge">
+                      intervals
+                    </EuiBadge>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </FormattedFormRow>
+            )}
+          </Field>
+
+          {/* ---------------- Sparse data handling ---------------- */}
+          <Field
+            name="imputationOption.imputationMethod"
+            id="imputationOption.imputationMethod"
+          >
+            {({ field, form }: FieldProps) => {
+              // If "Custom value" is selected, ensure at least one row is present
+              useEffect(() => {
+                if (
+                  field.value === SparseDataOptionValue.CUSTOM_VALUE &&
+                  (!form.values.imputationOption?.custom_value ||
+                    form.values.imputationOption.custom_value.length === 0)
+                ) {
+                  form.setFieldValue('imputationOption.custom_value', [
+                    { featureName: '', value: undefined },
+                  ]);
+                }
+              }, [field.value, form]);
+
+              return (
+                <>
+                  <FormattedFormRow
+                    fullWidth
+                    title="Sparse data handling"
+                    hint={[`Choose how to handle missing data points.`]}
+                    hintLink={FORECASTER_DOCS_LINK}
+                    isInvalid={isInvalid(field.name, form)}
+                    error={getError(field.name, form)}
+                  >
+                    <div style={{ maxWidth: FIELD_MAX_WIDTH }}>
+                      <EuiCompressedSelect 
+                        {...field} 
+                        options={sparseDataOptions}
+                        disabled={!isEditable}
+                      />
+                    </div>
+                  </FormattedFormRow>
+
+                  {field.value === SparseDataOptionValue.CUSTOM_VALUE && (
+                    <div style={{ maxWidth: FIELD_MAX_WIDTH }}>
+                      <EuiSpacer size="m" />
+                      <EuiText size="xs">
+                        <h5>Custom value</h5>
+                      </EuiText>
+                      <EuiSpacer size="s" />
+                      <FieldArray name="imputationOption.custom_value">
+                        {(arrayHelpers) => (
+                          <>
+                            {form.values.imputationOption.custom_value?.map(
+                              (_, index) => (
+                                <EuiFlexGroup
+                                  key={index}
+                                  gutterSize="s"
+                                  alignItems="center"
+                                >
+                                  <EuiFlexItem grow={false}>
+                                    <Field
+                                      name={`imputationOption.custom_value.${index}.data`}
```

Comment:
from line 258, it seems like we don't have `data` key in `imputationOption.custom_value`?